google/reader.go

```diff
@@ -26,9 +29,26 @@ func NewGcpReader(ctx context.Context, project, region, credentials string) (*GC
 		compute: comp,
 		project: project,
 		region:  region,
+		cache:   make(map[string]interface{}),
 	}, nil
 }
 
+func (r *GCPReader) setCache(key string, value interface{}) {
+	if _, ok := r.cache[key]; ok {
+		// key is already in the cache
```

Comment:
Unecessary comment, also shouldn't the old value be overwritten?

Reply:
You're right. If I work in many regions, we will have many `zones` with different values. IMHO, old value should not be overwritten, only the key should be more explicit. 

```
setCache("%s-zone", region)
```